crates/bevy_hierarchy/src/one_to_one/event.rs

```diff
@@ -0,0 +1,46 @@
+use core::marker::PhantomData;
+
+use bevy_ecs::{entity::Entity, event::Event};
+
+/// An [`Event`] that is fired whenever there is a change in [`OneToOne`] relationship `R`.
+///
+/// [`Event`]: bevy_ecs::event::Event
+#[derive(Event)]
+pub enum OneToOneEvent<R> {
```

Comment:
I'm not very happy with way I've defined these `Event` types, since they expose the `PhantomData` field. I'm not sure of an alternative though and would love some feedback!

Reply:
Maybe you could use a struct instead:
```rust
pub enum RelationEventType{
  Added,
  Removed,
}

pub struct OneToOneEvent<R>{
  phantom_data: PhantomData<fn(R)>,
  pub left: Entity,
  pub right: Entity,
  pub event_type: RelationEventType
}
```
Though I would probably not `pub` any field just provide constructors / accessors. They are the same size (24).

Reply:
Yeah I agree with this. I was hoping to avoid a struct because it makes pattern matching more annoying, but it's the better option here.